### PR TITLE
Add Claude response to "@!claude" in new or updated comments on issues or PRs

### PR DESCRIPTION
**Pull request by Matthew-Jennings**
This is a cool one! First pass for #1833. Check out Claude's response [here](https://github.com/Matthew-Jennings/pymedphys/pull/1#issuecomment-2011701910)!

A few things to consider (very open to ideas):

1. Which model? It's set to Haiku at the moment.
2. Is the `@!` syntax okay? Can anyone think of something prettier?
3. Should Claude quote the original comment in its response as it is now?
4. Haven't tested for issues (difficult when not in main)
5. What do you think of the response format overall?
6. Any other body params to consider tweaking? See [this link](https://docs.anthropic.com/claude/reference/messages_post) for reference. E.g.,
   - `system`
   - `stop_sequences`
   - `temperature`
   - `top_k`

### CODE REVIEW

**Comment by pchlap** [comment]
This is very cool! Great work @Matthew-Jennings. I think @! is as good as any option... I guess `@claude` would be cool but I suppose that might get mixed up with @-ing github users directly...

I'll let Simon comment on the model parameter specifics. I like that Claude quotes the original question to avoid confusion if lots of questions are being asked in the same PR. But this looks great overall, even if Clause isn't prepared to join as a PyMedPhys maintainer just yet 😆

**Comment by SimonBiggs** [comment]
https://github.com/Matthew-Jennings/pymedphys/pull/1#issuecomment-2013678963

**Comment by SimonBiggs** [comment]
Let's just use "Claude" without the @ or !

Also made another comment here:
https://github.com/Matthew-Jennings/pymedphys/pull/1#issuecomment-2013690205

File: .github/workflows/claude-assistant.yml
Comment (SimonBiggs) [inline .github/workflows/claude-assistant.yml:15]:
```suggestion
      run: echo comment=$(echo ${{github.event.comment.body}} | sed 's/claude//g') >> $GITHUB_OUTPUT
```

File: .github/workflows/claude-assistant.yml
Comment (SimonBiggs) [inline .github/workflows/claude-assistant.yml:29]:
```suggestion
                "model": "claude-3-opus-20240229",
```

File: .github/workflows/claude-assistant.yml
Comment (SimonBiggs) [inline .github/workflows/claude-assistant.yml:32]: Happy to merge at this, but do a follow up PR with all the issue history included

**Review by SimonBiggs — APPROVED** [review]

**Comment by SimonBiggs** [comment]
Regarding params we can leave them as they are now. They become more important when we start to do reflexion like workflows:

https://arxiv.org/abs/2303.11366

Regarding the prompt, feel free to put the first line of this in system, and then feed the remaining just into one single user message;

```
You are the AI assistant Claude, you are providing support on the PyMedPhys GitHub issue tracker. Please be helpful and supportive.

<github-issue-history>
Content from GitHub issue history API here
</github-issue-history>

<user-query>
Write in the most recent query here that pinged Claude the user sent
</user-query>
```

**Comment by SimonBiggs** [comment]
Oops, I merged without applying my suggestions :(

**Comment by SimonBiggs** [comment]
Other things @Matthew-Jennings, how hard do you think it would be to also include within the message sent off to Opus?

```
<files-included-in-pr>
</files-included-in-pr>

<diff-for-this-pr>
</diff-for-this-pr>
```

**Comment by Matthew-Jennings** [comment]
> Other things @Matthew-Jennings, how hard do you think it would be to also include within the message sent off to Opus?
> 
> ```
> <files-included-in-pr>
> </files-included-in-pr>
> 
> <diff-for-this-pr>
> </diff-for-this-pr>
> ```

I'll give it a red hot go, @SimonBiggs!

**Comment by SimonBiggs** [comment]
Also... If all off the above worked... The follow up version to that would be for Claude to also get a record of test failures, and then creates a PR into the branch in question that attempts to fix the CI failures...

**Comment by SimonBiggs** [comment]
Thanks Claude! You're a gem :)

**Pull request merged**
